src/test/store/profile-view.test.js

```diff
@@ -609,6 +609,17 @@ describe('actions/ProfileView', function() {
     });
   });
 
+  describe('changeNetworkSearchString', function() {
+    it('changes the search string', function() {
+      const profile = getProfileWithMarkers([['a', 0, null], ['b', 1, null]]);
+      const { dispatch, getState } = storeWithProfile(profile);
+
+      expect(UrlStateSelectors.getNetworkSearchString(getState())).toEqual('');
+      dispatch(ProfileView.changeNetworkSearchString('a'));
+      expect(UrlStateSelectors.getNetworkSearchString(getState())).toEqual('a');
```

Comment:
This test is good as it checks that we are remembering the search string. It would be good to do another test to check that we are actually filtering the network markers correctly. This would be good as a second `it` block.

Reply:
Here is what I'm thinking here, let me know if you need more clarification.

```js
describe('changeNetworkSearchString', function() {
 it('changes the search string', function() {
   const profile = getNetworkTrackProfile();
   const { dispatch, getState } = storeWithProfile(profile);

   expect(UrlStateSelectors.getNetworkSearchString(getState())).toEqual('');
   dispatch(ProfileView.changeNetworkSearchString('a'));
   expect(UrlStateSelectors.getNetworkSearchString(getState())).toEqual('a');
 });

 it('filters the network markers', function() {
   const profile = getNetworkTrackProfile();
   const { dispatch, getState } = storeWithProfile(profile);

   // The following lines are pseudo-code. Replace it with correct code to make the
   // test work.
   expect(getNetworkMarkers().length).toEqual(10);
   dispatch(ProfileView.changeNetworkSearchString('something that matches a marker'));
   expect(getNetworkMarkers().length).toEqual(1);
   expect(getNetworkMarkers()[0].name).toEqual('something that matches a marker');
 });
});

```

Reply:
@gregtatum ,
Thanks for the suggestion.
But I am a bit confused about [getNetworkMarker()](https://github.com/firefox-devtools/profiler/blob/master/src/test/fixtures/profiles/processed-profile.js#L499-L523) function. It returns an array of two markers no matter what! It also doesn't use `networkSearchString` anywhere. Then how to test is whether markers are getting filtered by `networkSearchString` or not?
Are you talking about the [getNetworkMarker()](https://github.com/firefox-devtools/profiler/blob/master/src/selectors/per-thread/markers.js#L189-L192) **selector**?

Reply:
hey @rajmeghpara, yeah Greg talks about the selector `getNetworkMarkers` :) (also notice the final `s`).